mpich 3.4 doesn't build with clang 6.0.0

### Steps to reproduce the issue

Ping @raffenet @yfguo, can you reproduce? Should we add a conflict for clang?

```console
$ spack install mpich~fortran target=x86_64 %clang@6.0.0
```

### Error Message

```console
libtool: link: ranlib lib/.libs/libmpi.a
libtool: link: rm -fr lib/.libs/libmpi.lax lib/.libs/libmpi.lax
libtool: link: ( cd "lib/.libs" && rm -f "libmpi.la" && ln -s "../libmpi.la" "libmpi.la" )
/bin/bash ./libtool  --tag=CXX   --mode=link /opt/spack/lib/spack/env/clang/clang++  -O2 -version-info 13:10:1 -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/libfabric-1.11.0-6azfrqsjfn3bthebhhkb7b2tfztchbfs/lib -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/hwloc-2.4.0-eaeek6a7vkgryl5pciiuloro3txts4df/lib -o lib/libmpicxx.la -rpath /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/mpich-3.4.1-53plqekl5t7h5cfz6jkpmrxzzvirxvpt/lib src/binding/cxx/initcxx.lo lib/libmpi.la 
/bin/bash ./libtool  --tag=CC   --mode=link /opt/spack/lib/spack/env/clang/clang -fvisibility=hidden  -O2  -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/libfabric-1.11.0-6azfrqsjfn3bthebhhkb7b2tfztchbfs/lib -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/hwloc-2.4.0-eaeek6a7vkgryl5pciiuloro3txts4df/lib -o src/env/mpichversion src/env/mpichversion.o lib/libmpi.la 
/bin/bash ./libtool  --tag=CC   --mode=link /opt/spack/lib/spack/env/clang/clang -fvisibility=hidden  -O2  -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/libfabric-1.11.0-6azfrqsjfn3bthebhhkb7b2tfztchbfs/lib -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/hwloc-2.4.0-eaeek6a7vkgryl5pciiuloro3txts4df/lib -o src/env/mpivars src/env/mpivars.o lib/libmpi.la 
libtool: link: /opt/spack/lib/spack/env/clang/clang -fvisibility=hidden -O2 -o src/env/.libs/mpichversion src/env/mpichversion.o  -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/libfabric-1.11.0-6azfrqsjfn3bthebhhkb7b2tfztchbfs/lib -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/hwloc-2.4.0-eaeek6a7vkgryl5pciiuloro3txts4df/lib lib/.libs/libmpi.so -lhwloc -lfabric -latomic -ldl -lpthread -lrt -Wl,-rpath -Wl,/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/mpich-3.4.1-53plqekl5t7h5cfz6jkpmrxzzvirxvpt/lib
libtool: link: /opt/spack/lib/spack/env/clang/clang -fvisibility=hidden -O2 -o src/env/.libs/mpivars src/env/mpivars.o  -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/libfabric-1.11.0-6azfrqsjfn3bthebhhkb7b2tfztchbfs/lib -L/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/hwloc-2.4.0-eaeek6a7vkgryl5pciiuloro3txts4df/lib lib/.libs/libmpi.so -lhwloc -lfabric -latomic -ldl -lpthread -lrt -Wl,-rpath -Wl,/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/clang-6.0.0/mpich-3.4.1-53plqekl5t7h5cfz6jkpmrxzzvirxvpt/lib
lib/.libs/libmpi.so: undefined reference to `__gnu_f2h_ieee'
lib/.libs/libmpi.so: undefined reference to `__gnu_h2f_ieee'
```

### Information on your system


    Spack: 0.16.0-997-aa8e026242
    Python: 3.8.7
    Platform: linux-ubuntu20.04-zen2
    Concretizer: clingo


### Additional information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output

--- a/var/spack/repos/builtin/packages/mpich/package.py
+++ b/var/spack/repos/builtin/packages/mpich/package.py
@@ -187,6 +187,9 @@ spack package at this time.''',
     conflicts('+pci', when='@:3.2~hydra')
     conflicts('+libxml2', when='@:3.2~hydra')
 
+    # see https://github.com/pmodels/mpich/pull/5030
+    conflicts('%clang@:7', when='@3.4:')
+
     @classmethod
     def determine_version(cls, exe):
         output = Executable(exe)(output=str, error=str)

--- a/var/spack/repos/builtin/packages/mpich/package.py
+++ b/var/spack/repos/builtin/packages/mpich/package.py
@@ -187,7 +187,7 @@ spack package at this time.''',
     conflicts('+pci', when='@:3.2~hydra')
     conflicts('+libxml2', when='@:3.2~hydra')
 
-    # see https://github.com/pmodels/mpich/pull/5030
+    # see https://github.com/pmodels/mpich/pull/5031
     conflicts('%clang@:7', when='@3.4:')
 
     @classmethod

